File: cbl-getting-started/obj-c/ViewController.m
#import "ViewController.h"

#import <CouchbaseLite/CouchbaseLite.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    NSError *error;
    CBLDatabase *database = [[CBLDatabase alloc] initWithName:@"mydb" error:&error];

    // ドキュメント作成
    CBLMutableDocument *mutableDoc = [[CBLMutableDocument alloc] init];
    [mutableDoc setString:@"佐藤" forKey:@"lastname"];
    [mutableDoc setString:@"user" forKey:@"type"];

    // ドキュメント保存
    [database saveDocument:mutableDoc error:&error];

    // ドキュメント取得、変更、保存
    CBLMutableDocument *mutableDoc2 =
        [[database documentWithID:mutableDoc.id] toMutable];
    [mutableDoc2 setString:@"太郎" forKey:@"firstname"];
    [database saveDocument:mutableDoc2 error:&error];


    CBLDocument *document = [database documentWithID:mutableDoc2.id];
    // ドキュメントIDとプロパティの出力
    NSLog(@"ドキュメントID: %@", document.id);
    NSLog(@"タイプ: %@", [document stringForKey:@"type"]);
    NSLog(@"ユーザー名: %@ %@", [document stringForKey:@"lastname"], [document stringForKey:@"firstname"]);

    // クエリ作成
    CBLQueryExpression *type =
        [[CBLQueryExpression property:@"type"] equalTo:[CBLQueryExpression string:@"user"]];
    CBLQuery *query = [CBLQueryBuilder select:@[[CBLQuerySelectResult all]]
                                        from:[CBLQueryDataSource database:database]
                                        where:type];

    // クエリ実行
    CBLQueryResultSet *result = [query execute:&error];
    NSLog(@"ユーザー数: %lu", (unsigned long)[[result allResults] count]);

    // データベースクローズ
    if (![database close:&error])
        NSLog(@"Error closing db:%@", error);

}


@end

File: cbl-getting-started/swift/ViewController.swift
import UIKit
import CouchbaseLiteSwift

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        getStarted()
    }

    func getStarted () {
        // データベース作成または取得（再実行時）
        let database: Database
        do {
            database = try Database(name: "mydb")
        } catch {
            fatalError("Error opening database")
        }

        // ドキュメント作成
        let mutableDoc = MutableDocument().setString("佐藤", forKey: "lastname")
            .setString("user", forKey: "type")

        // ドキュメント保存
        do {
            try database.saveDocument(mutableDoc)
        } catch {
            fatalError("Error saving document")
        }

        // ドキュメント取得、変更、保存
        if let mutableDoc = database.document(withID: mutableDoc.id)?.toMutable() {
            mutableDoc.setString("太郎", forKey: "firstname")
            do {
                try database.saveDocument(mutableDoc)

                let document = database.document(withID: mutableDoc.id)!
                print("ドキュメントID: \(document.id)!)")
                print("名前： \(document.string(forKey: "lastname")!) \(document.string(forKey: "firstname")!)")
            } catch {
                fatalError("Error updating document")
            }
        }

        // クエリ
        print("クエリ実行")
        let query = QueryBuilder
            .select(SelectResult.all())
            .from(DataSource.database(database))
            .where(Expression.property("type").equalTo(Expression.string("user")))

        do {
            let result = try query.execute()
            print("ユーザー数: \(result.allResults().count)")
        } catch {
            fatalError("Error running the query")
        }

        do {
            try database.close()
        } catch {
            fatalError("Error running the query")
        }
    }
}
